templates/zerver/api/incoming-webhooks-walkthrough.md

```diff
@@ -36,6 +36,38 @@ thing, it requires only one fixture,
 }
 ```
 
+If your integration depends on HTTP headers sent by the service you are integrating with,
+then it would be a good idea to store these too. An example of when you would need to do
+this would be if the service sends the event type as a custom http header, like with
+GitHub's webhooks.
+
+To do this,
+
+1. In the folder where your integration's code lives, create a file callled
+`headers.py`.
+
+2. Inside this file, create a python dictionary named `fixtures_to_headers_map`. This
+dictionary should map a fixture's name to a dictionary of headers that you'd like
+to associate the fixture with. Alternatively, the fixture's name could also map to a
+function which should in turn generate a dictionary of headers. Mapping to a function
+would be useful if you need to do some extra processing or contact a third party
+endpoint or etc.
```

Comment:
Er, a single fixture doesn't have multiple headers, so this is backwards. Usually, an event type (header) will map to one or more fixtures. In some cases, an event type might be ignored and might not have any fixtures at all. Does that make sense?

Reply:
What I did here was add considerations to potential future requirements. While we don't have any such integrations like this yet, we might require more than one header (i.e. more than just an event type header) in future integrations (even though the chances of this are rare, they're still non-zero).

Also, performing a reverse mapping from headers to fixtures is more difficult to work with. Generally, we have a fixture name to start with, and _then_ we can try to find the headers corresponding to it in both the testing framework and the IDP.

Reply:
I think it's reasonable to expect that some future integration will expect to set multiple HTTP headers.  But in the architecture I have proposed (with details below), that's easy -- we just make the `fixture_name_to_headers` function more complicated than the basic one I wrote for GitHub.